src/Models/Refund.php

```diff
@@ -38,12 +39,11 @@ public static function boot()
         });
 
         static::saving(function ($refund) {
-            if (empty($refund->payment_id)) {
-                throw new \Exception('A refund must be applied to a payment.');
-            }
-            if (empty($refund->amount)) {
-                throw new \Exception('A refund must be for an amount.');
-            }
+            Assert::lazy()
+                ->that($refund->payment_id)->notEmpty('A refund must be applied to a payment.')
+                ->that($refund->amount)->notEmpty('A refund must be for an amount.')
+                ->verifyNow();
+            //Todo: Add the following to the assert if possible
             if ($refund->amount > ($refund->payment->amount_refundable)) {
```

Comment:
@pdbreen is there an assert method that I could use to to evaluate a conditional statement like the one below?

Reply:
`->that($refund->amount > ($refund->payment->amount_refundable))->false('message')`

Reply:
Might be better to use `lessOrEqualThan`  -- I think like this, but not at my desk so can't confirm exact syntax

 `->that($refund->amount)->lessOrEqualThan($refund->payment->amount_refundable, 'message')`

https://github.com/beberlei/assert#list-of-assertions

Reply:
Thank you!